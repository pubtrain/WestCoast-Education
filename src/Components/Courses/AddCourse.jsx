import { useRef, useContext } from 'react';
import { v4 as uuidv4 } from 'uuid';
import AppContext from '../Store/app-context';

const AddCourse = ({ closeModal }) => {
  const courseTitleInputRef = useRef();
  const courseLengthInputRef = useRef();
  const courseDescriptionInputRef = useRef();
  const courseStartingDateInputRef = useRef();
  const formResetRef = useRef();
  const context = useContext(AppContext);

  const onSubmitHandler = (e) => {
    e.preventDefault();

    const title = courseTitleInputRef.current.value;
    const length = courseLengthInputRef.current.value;
    const description = courseDescriptionInputRef.current.value;
    const startingDate = courseStartingDateInputRef.current.value;

    const body = {
      id: uuidv4(),
      title,
      length,
      description,
      startingDate,
    };

    context.addCourse(body);
    formResetRef.current.reset();

    closeModal(!true);
    window.location.reload();
  };

  return (
    <form onSubmit={onSubmitHandler} ref={formResetRef}>
      <h2>Lägg till ny kurs</h2>
      <div className="form-control">
        <label htmlFor="courseTitle"></label>
        <input
          type="text"
          id="courseTitle"
          placeholder="Kurstitel"
          ref={courseTitleInputRef}
        />
      </div>
      <div className="form-control">
        <label htmlFor="courseLength"></label>
        <input
          type="text"
          id="courseLength"
          placeholder="Kurslängd"
          ref={courseLengthInputRef}
        />
      </div>
      <div className="form-control">
        <label htmlFor="courseDescription"></label>
        <textarea
          id="courseDescription"
          placeholder="Kursbeskrivning"
          ref={courseDescriptionInputRef}
        />
      </div>
      <div className="form-control">
        <label htmlFor="courseStartingDate"></label>
        <input
          type="text"
          id="courseStartingDate"
          placeholder="Kursen startar"
          ref={courseStartingDateInputRef}
        />
      </div>
      <button>Lägg till kurs</button>
    </form>
  );
};

export default AddCourse;
