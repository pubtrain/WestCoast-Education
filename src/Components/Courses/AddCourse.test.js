import React from 'react';
import { render, screen } from '@testing-library/react';
import AddCourse from './AddCourse';
import { MemoryRouter } from 'react-router-dom';

describe('AddCourse form', () => {
  const setup = () => render(<AddCourse />);
  test('course form renders correctly', () => {
    setup();
    const utils = render(<AddCourse />, { wrapper: MemoryRouter });
    expect(utils).toMatchSnapshot();
    expect(utils).toBeTruthy();
  });
});
