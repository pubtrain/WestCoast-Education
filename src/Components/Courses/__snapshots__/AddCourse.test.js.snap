// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCourse form course form renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <h2>
          Lägg till ny kurs
        </h2>
        <div
          class="form-control"
        >
          <label
            for="courseTitle"
          />
          <input
            id="courseTitle"
            placeholder="Kurstitel"
            type="text"
          />
        </div>
        <div
          class="form-control"
        >
          <label
            for="courseLength"
          />
          <input
            id="courseLength"
            placeholder="Kurslängd"
            type="text"
          />
        </div>
        <div
          class="form-control"
        >
          <label
            for="courseDescription"
          />
          <textarea
            id="courseDescription"
            placeholder="Kursbeskrivning"
          />
        </div>
        <div
          class="form-control"
        >
          <label
            for="courseStartingDate"
          />
          <input
            id="courseStartingDate"
            placeholder="Kursen startar"
            type="text"
          />
        </div>
        <button>
          Lägg till kurs
        </button>
      </form>
    </div>
    <div>
      <form>
        <h2>
          Lägg till ny kurs
        </h2>
        <div
          class="form-control"
        >
          <label
            for="courseTitle"
          />
          <input
            id="courseTitle"
            placeholder="Kurstitel"
            type="text"
          />
        </div>
        <div
          class="form-control"
        >
          <label
            for="courseLength"
          />
          <input
            id="courseLength"
            placeholder="Kurslängd"
            type="text"
          />
        </div>
        <div
          class="form-control"
        >
          <label
            for="courseDescription"
          />
          <textarea
            id="courseDescription"
            placeholder="Kursbeskrivning"
          />
        </div>
        <div
          class="form-control"
        >
          <label
            for="courseStartingDate"
          />
          <input
            id="courseStartingDate"
            placeholder="Kursen startar"
            type="text"
          />
        </div>
        <button>
          Lägg till kurs
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <h2>
        Lägg till ny kurs
      </h2>
      <div
        class="form-control"
      >
        <label
          for="courseTitle"
        />
        <input
          id="courseTitle"
          placeholder="Kurstitel"
          type="text"
        />
      </div>
      <div
        class="form-control"
      >
        <label
          for="courseLength"
        />
        <input
          id="courseLength"
          placeholder="Kurslängd"
          type="text"
        />
      </div>
      <div
        class="form-control"
      >
        <label
          for="courseDescription"
        />
        <textarea
          id="courseDescription"
          placeholder="Kursbeskrivning"
        />
      </div>
      <div
        class="form-control"
      >
        <label
          for="courseStartingDate"
        />
        <input
          id="courseStartingDate"
          placeholder="Kursen startar"
          type="text"
        />
      </div>
      <button>
        Lägg till kurs
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
